tests/integration/test_charm.py

```diff
@@ -139,3 +140,42 @@ async def test_s3_storage(
     )
     with open(downloaded_test_file, "r", encoding="utf-8") as test_fd:
         assert test_content in test_fd.read()
+
+
+async def test_scale_workload(
+    ops_test: OpsTest,
+    app: Application,
+    kube_core_client,
+):
+    """
+    arrange: after charm deployed and ready.
+    act: scale application to 3 units and kill the current leader.
+    assert: the application should be reachable.
+    """
+
+    # get the pod name of the first unit (the leader)
+    model_name = list(ops_test.models.values())[0].model_name
+    pods = kube_core_client.list_namespaced_pod(namespace=model_name).items
+    # the alphabet used by k8s is restricted to avoid bad words
+    # https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/rand/rand.go#L83
+    k8s_pod_ending_regex = '[bcdfghjklmnpqrstvwxz2456789]{9,10}-[bcdfghjklmnpqrstvwxz2456789]{5}'
+    mattermost_pod_regex = re.compile('mattermost-k8s-' + k8s_pod_ending_regex)
+    leader_pod = next(
+        p.metadata.name for p in pods if re.match(mattermost_pod_regex, p.metadata.name)
+    )
+
+    # scale the application
+    await ops_test.juju("scale-application", "mattermost-k8s", "3")
+    await ops_test.model.wait_for_idle(status="active")
+
+    # kill the leader
+    kube_core_client.delete_namespaced_pod(name=leader_pod, namespace=model_name)
+    await ops_test.model.wait_for_idle(status="active")
+
+    unit_informations = json.loads(
+        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
+    )
+    mattermost_ip = unit_informations[app.units[0].name]["address"]
```

Comment:
Would the `public_address` of the unit work? https://pythonlibjuju.readthedocs.io/en/latest/api/juju.unit.html

Reply:
It doesn't work, that returns the IP address of my laptop on my local network (using microk8s)